### ISSUE TITLE
UI: RFC: splash screen warning

### ISSUE THREAD

**Issue by dexX7** [zathras-crypto/omnicore#37]
Right now, on each start, a "scary" warning is shown. While I believe a warning in general isn't a bad thing, I'm wondering about:
1. Should this warning be shown on mainnet, testnet and regtest or only mainnet?
2. Should this warning be refined, e.g. with softer language?
3. Should this warning be shown on _every_ start or only once, or once, mainnet and testnet?

Ping @oktype as UIX expert. :)

On Ubuntu 14.04.01 with QT4:
![splash_ubuntu](https://cloud.githubusercontent.com/assets/5836089/6318993/e1c71544-baac-11e4-8fa4-2eba9f6565cd.png)

On Windows 8.1:
![splash_windows](https://cloud.githubusercontent.com/assets/5836089/6318994/e34abb6e-baac-11e4-9217-3eadbd5d7578.png)

**Comment by oktype** [comment]
**WARNING - THIS IS ONLY MY TAKE ON THE ISSUE AND MY ADVICE SHOULD NOT BE CONSIDERED AS A LEGAL ADVICE**
This is mostly a legal issue, so I guess that the amount of focus and the language this notification contain should be checked first from that perspective (especially If the purpose of this notification is also to actually try and remove responsibility from the foundation/developers) 

Set aside the obligations, we still want to inform our users in a responsible way. So with that in mind, I would suggest: Yes, the language should be refined.
Right now those paragraphs can get 2 results: either the user will stay away from the platform and never come back, or he will try to categorize this new information only as an (somewhat over-excessive) effort to remove responsibility from the foundation/developers, hence he won't treat it with the respect/attention it should get. 
Not too sure what are our possibilities for that notification box design-wise

also, If possible, I think that making sure the user read this notification at least once is enough (even use a 5 sec timer if possible). while adding some "under development" prominent link from within the UI leading to that notification for later reads (same as on Omni Wallet)

So:
1.I'm not sure I'm fully up to date with the differences testnet/regtest/mainnet and the capabilities to spend real value on each so can't advise on that issue
2.in my opinion: yes.
3.make sure the user reads it once, then give him access to read it again when he want to ponder in the meaning of 'under development'

**Comment by oktype** [comment]
From my humble experience with legal stuff, it is best to not try to make it look legal.
Make the shortest and easiest text to read and understand. (not a legal advice)

**Comment by dexX7** [comment]
The project is licsensed as MIT open source project and the warning is sort of similar, see: http://opensource.org/licenses/MIT.

I think the main goal here is to raise awareness with the underlying message that a) it's no bullet proofed software release (even though we're very restrictive and it should not result in loss of coins), and because the protocol is updated from time to time, a user should at least know, that an update may be required at some point.

> Not too sure what are our possibilities for that notification box design-wise

Currently it's only a plain dialog box, but styling or changing this window shouldn't be very difficult, and it's basically drag and drop ([see here](http://i.imgur.com/uKN9wOx.png)).

> 1. I'm not sure I'm fully up to date with the differences testnet/regtest/mainnet and the capabilities to spend real value on each so can't advise on that issue

Regtest is a "local blockchain mode" for developers with completely isolated environment, where bitcoins can be generated on by the user. This is used to test applications. In my opinion: no warning at all required.

Testnet is a "playground". "Test" bitcoins have no value and it's basically as "public testing environment". _Imho_ no warning required as well.

Mainnet is "real", with "real" bitcoins, and potential loss of monetary value, if something goes very, very wrong.

> 1. make sure the user reads it once, then give him access to read it again when he want to ponder in the meaning of 'under development'

I like the idea of providing a choice to open this warning once again. Maybe from within the "About Omni Core" dialog, which can be accessed via the menu? It looks a bit broken at the moment and also needs a refinement, but this seems like the way to go. Maybe we could simply copy the message and show it there as well:

![about](https://cloud.githubusercontent.com/assets/5836089/6319454/02e2f8dc-babd-11e4-9cb5-2859367937c7.png)

Is there any landing page available for users? It states the software is under active developmenet, but what's really missing is any information _where_ the user could get additional information about the project or updated. What should he or she be referred to?

**Comment by oktype** [comment]
-The decision of where to put the notification link how much we want to emphasize the warning.
Omni wallet also contains this warning as a prominent link from the footer (visible from every page).
-Depends on the length of the text, it could be squeezed into the about dialog, or as a separate box, also available from a link and a short description from the about box.

I like the 'under development' definition much more then the 'experimental' one since it holds more potential and imply on the development efforts , but still imply it is not the final product.

As for the regtest/testnet/mainnet. if the risk warning is only relevant to the mainnet environment, warn only the users that uses this environment. 'test' imply the experimental feature, and since there is no risk, there is no reason for a warning.

We should try to find a better way to communicate 'Mainnet' for the users as this is a developers term.
Mainnet should be considered as the actual product, So when you are not working on the test environments (which should be hinted) you are using the product. (I might be wrong with this since the Omni Core target audience is tec savvy)

I agree with the fact that a documentation is needed and that the project should make information as approachable as possible. I've suggested to make some kind of help pages 'info center' as part of Omni. there is also a already implemented pending more robust website ready to be published instead of the current omnilayer.org landing page, as soon as it will be filled with the suitable content.
If currently there is no other place to get such information, maybe we should rush such space.
@achamely @CraigSellars

**Comment by dexX7** [comment]
> We should try to find a better way to communicate 'Mainnet' ...

It could be replaced by something that referrs to "production environments".

I'm not suitable to rephrase the text, but I'd be happy to replace it.

In general: I'd try to avoid causing "fear" whenever possible and chose a friendly language. Instead of starting with ALL-CAPS-WARNINGS, the user should be welcomed and informed. Keep it positive, it's the very first impression right after starting the client that is created! :)

> I agree with the fact that a documentation is needed
> I've suggested to make some kind of help pages 'info center' as part of Omni.
> If currently there is no other place to get such information, maybe we should rush such space.

I would not recommend to build a help center within the application itself, if this is what you were referring to, but instead providing some kind of documentation or landing webpage.

At minimum there should be landing page, which states something along the lines:

```
The current Omni Core version is vX.Y.Z.

Users with previous versions must update their client until *some-date*, or otherwise ...
```

However simple this landing page might be, it's very, very, very crucial in my opinion. And I'd really like to provide a link to such a page from within Omni Core.

Quick reasoning why this is so important @oktype: whenever there is a protocol update (and major client update), say for example when a new transaction type is enabled, like the distributed token exchange, users with _outdated_ clients basically can end up with a very different "view of world", and it's very likely that those users would see completely different and wrong balances.

Big slap @CraigSellars by the way (but sorry, if you're not the one to blame) for not publishing any post on the blog, the forums and mailing lists, mentioning the "send to owners" enabling at all. I showed once that I was able to drain masterxchange, because I exploited exactly this, and this is something that _all_ users of Omni Core (and others interacting with the Omni Layer) _must_ be informed of...

---

To summarize the discussion so far:
- [ ] No warning should be shown for testnet and regtest mode, but mainnet only, because this is where real monetary value is at stake.
- [ ] The warning should be shown once, but not every time the client is started.
- [ ] From within the application there should be an option/menu item to show the warning again.
- [ ] The warning should be rephrased.
- [ ] There should be a landing page with further information, or at minimum some information about the current release and whether an update for the client is available or required.

I can take care of 1, 2, 3, but need someone else for 4 + 5.

Quick question: is there any contact email available for support, bugs reports, ...? Right now there are a few places where `info@bitcoin.org` is shown, but I'm pretty sure we should replace this.

**Comment by zathras-crypto** [comment]
Hey guys,

Just wanted to add I stand by a very scary warning.  If anything I stand by making it even _more_ scary.

For me this ties into the default wallet discussion.  Omni Core is unlike any other 'alt'coin/wallet.  By using our software you are not just talking about your Omni tokens, but also putting all of your Bitcoins in Omni's hands by default (a situation I am still against).

Example scenario - I have 200 Bitcoins in my local wallet.  I download and run OmniCore.  I thus put all of those Bitcoins at the mercy of a bug in OmniCore (and for Joe Q User may not have any clue that OmniCore even has access to their Bitcoins without such a warning).  Thus, yep I do kind of want to really drive the point home that this is extremely new software and somewhat experimental.  We're effectively asking users to put their Bitcoins in the hands of code that has never seen end user usage before & trust that everything is going to be OK.

Now for those of us with half a brain, 200 BTC should be in cold storage not in a local QT wallet but I use it as the example to drive the point home - while there is no segregation of Bitcoin and Omni wallet files and no tangible history of end user usage to give us N level of confidence in its safety, is not the responsible thing to err on the side of caution, and make sure the user knows all this?

It's been on the back burner, but FYI I'd intended to push for an expansion of the warning (since it doesn't actually make it clear enough at all in my book that your Bitcoins are accessible to and will by used by OmniCore).  We just talk about 'wallets', and for the regular end user they may well assume just like pretty much any other QT wallet out there, that OmniCore uses its own wallet file which is not the case.

TL:DR; separate the wallets and I wouldn't care if we slimmed the message to a single line, but as long as we insist on using the users existing Bitcoin wallet for the sake of 'convenience', I'll always be on the side of making sure users know what's up - it's the responsible thing to do IMO.

Open to debate of course :) but being open, transparent, truthful and making sure our users have all the information to hand is of critical importance to me.

Thanks
Z

**Comment by dexX7** [comment]
@zathras-crypto: I think the general consensus is that a warning is very reasonable for the reasons you stated. "Scary" was maybe not the right word. The intention is to make sure the user knows what he or she does, namely risking coins, and what I was referring to was the tone of delivering this message. My point is basically about finding a balance between really making sure the user understands and creating a positive user experience.

Here is a similar example to which context I'm referring to: https://github.com/mastercoin-MSC/omniwallet/issues/895

I'd be very curious about hearing your opinion about a) whether the warning should be shown for testnet and regtest, and b) whether it should be shown once or every time the client is started.

**Comment by zathras-crypto** [comment]
> I'd be very curious about hearing your opinion about a) whether the warning should be shown for testnet and regtest, and b) whether it should be shown once or every time the client is started.

Not fussed about testnet/regtest there is no risk to end user funds there, and FYI I was instructed a while ago to code it so it shows once at first run and then is never shown again (I just haven't prioritized that to date).

I appreciate what you're saying, I guess the two main points I'm trying to get across are:
1) Nowhere at all in the message does it state we will use the users existing Bitcoin wallet.  A new user would not be informed that he/she is trusting their Bitcoins to our code.  This to me is unacceptable.
2) We should not provide any false senses of stability/security.  We should be transparent that this is brand new stuff and could go wrong.

I'm absolutely open to changed wording, but the core 'message' if you will needs to be communicated.

Hope that clarifies a bit :)

Thanks
Z

**Comment by zathras-crypto** [comment]
P.S. for anyone in this discussion that isn't aware, for the 'background' that warning is a direct copy and paste from an older project called the Masterchest Wallet and was written by yours truly _intentionally_ to be scary (because I had quite a stressful time during/just after the maidsafe crowdsale watching people pump millions of dollars through a piece of software I'd written seemingly not caring about the risks).  Just FYI :)

**Comment by zathras-crypto** [comment]
![image](https://cloud.githubusercontent.com/assets/5633351/6320581/13d7664a-bb36-11e4-9019-9489f99945d2.png)
Apologies, making a liar out of myself - seems has been reworded quite a bit since then...

**Comment by dexX7** [comment]
Alright then, I'm going to address the mainnet-only and "show once" part. This is something I have tested earlier.

Another idea: there could be a third option besides "Exit" and "Continue": (in the sense of) "Continue, but show the warning on next start again". Not sure how to style this though, as a dialog with three buttons and long text would probably look horrible.

>  I guess the two main points I'm trying to get across are ...

Sounds fine to me!

In addition, but this is probably not that easy: there could be an option to create a wallet.dat backup before the start. The (huge) downside is that wallet.dat backups were not only once misunderstood, because the file actually only holds 100 pre-generated keys per default, but once they are used, the backup wouldn't have the keys for the destinations where the coins might be held.

For the warning text itself, a part which highlights that Omni Core probably uses the same datadir and wallet.dat of an earlier installation, and that there could be direct exposure of those files, would be great. As general warning, even though I'm not really a fan of all-caps, the MIT license provides a blueprint:

```
THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
THE SOFTWARE.
```

**Comment by zathras-crypto** [comment]
> The (huge) downside...

We can prompt the user for their password and up the keypool to whatever (eg 10,000) we like before backup but that's not necessarily where I see the risk (there is nowhere we interact directly with the wallet file ourselves, that's all left to underlying bitcoin code) so I'm not so concerned with corruption as I am accidentally giving the users BTC to a miner or exodus/reference/multisig output etc - if that happens a wallet backup isn't going to help sadly.

> Another idea: there could be a third option besides "Exit" and "Continue": (in the sense of) "Continue, but show the warning on next start again". Not sure how to style this though, as a dialog with three buttons and long text would probably look horrible.

From a UX perspective, the vanilla way to handle this is to have the "Exit" and "Continue" buttons and then chuck a checkbox on left align saying "Do not show this message again".

> For the warning text itself...

It's still my honest belief that when we warn (which we absolutely must do) users that their existing Bitcoin are exposed to and used by our software, we will put some users off using it.  I still believe the best way to handle that is with a segregated wallet file (which can be overridden to use the existing bitcoin wallet.dat with one simple cmdline param for advanced users) but I also don't want to get us stuck going round in circles on an issue that has been discussed before (and IIRC didn't really get a definitive outcome).

Thanks
Z

**Comment by dexX7** [comment]
This was now addressed by #38, but awaiting a reworded warning, so let's keep it open.

**Comment by dexX7** [comment]
I'd really like to include a link to a landing page with version and update information etc., but this seems to be out of this scope.
